### PR TITLE
Add troubleshooting pages

### PR DESCRIPTION
**Pull request by underyx**

### CODE REVIEW

File: docs/troubleshooting/app.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Troubleshooting Semgrep App
+
+[TOC]
+
+# A project is reporting the last scan 'never started'
```
Comment (pabloest) [inline docs/troubleshooting/app.md:5]:
```suggestion
# A project reports the last scan 'never started'
```

File: docs/troubleshooting/app.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+## If a Semgrep CI job exists
+
+In this case, check the log output for any hints about what the issue is.
+
+- If the logs mention a missing token or an authentication failure,
+  you can get a new token from the Settings page of Semgrep App,
+  and set it as `SEMGREP_APP_TOKEN` in your CI provider's secret management UI.
+
+# A project is reporting the last scan 'never completed'
```
Comment (pabloest) [inline docs/troubleshooting/app.md:28]:
```suggestion
# A project reports the last scan 'never completed'
```

File: docs/troubleshooting/app.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+In this case the issue is likely with the CI configuration.
+
+- Make sure that the branch you committed a CI job to
+  is included in the list of branches the job is triggered on.
+- Make sure that the CI configuration file has valid syntax.
+  Most providers have a tool for checking the syntax of configuration files.
+
+## If a Semgrep CI job exists
+
+In this case, check the log output for any hints about what the issue is.
```
Comment (pabloest) [inline docs/troubleshooting/app.md:22]:
```suggestion
Check the log output for any hints about what the issue is.
```

File: docs/troubleshooting/app.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+# A project is reporting the last scan 'never started'
+
+This status means that your CI job never authenticated to Semgrep App.
+
+Check your CI provider (such as GitHub Actions) for the latest Semgrep job execution.
+
+## If you cannot find a Semgrep CI job
+
+In this case the issue is likely with the CI configuration.
```
Comment (pabloest) [inline docs/troubleshooting/app.md:13]:
```suggestion
The issue is likely with the CI configuration.
```

File: docs/troubleshooting/app.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  is included in the list of branches the job is triggered on.
+- Make sure that the CI configuration file has valid syntax.
+  Most providers have a tool for checking the syntax of configuration files.
+
+## If a Semgrep CI job exists
+
+In this case, check the log output for any hints about what the issue is.
+
+- If the logs mention a missing token or an authentication failure,
+  you can get a new token from the Settings page of Semgrep App,
```
Comment (pabloest) [inline docs/troubleshooting/app.md:25]: How about linking to the settings page at https://semgrep.dev/manage/settings ?

File: docs/troubleshooting/gitlab-sast.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+The Semgrep SAST job should take less than a minute
+to scan a large project with 50k lines of Python and TypeScript code.
+If you see worse performance,
+please [reach out](../support.md) to the Semgrep maintainers for help with tracking down the cause.
+Long runtimes are typically caused by just one rule or source code file taking too long.
+
+If you suspect large and complex source code files (such as minified JS or generated code)
+are making the job last too long, you might want to exclude these files from scanning.
+[GitLab's path exclusion feature](https://docs.gitlab.com/ee/user/application_security/sast/#vulnerability-filters)
+does not skip scanning excluded files. It scans everything, and then hides results from excluded files.
```
Comment (pabloest) [inline docs/troubleshooting/gitlab-sast.md:21]: Is there no workaround for that in GitLab SAST, other than switching over to "direct" use of Semgrep CI in GitLab as described below?

File: docs/troubleshooting/gitlab-sast.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+To improve performance by 10x on a typical project, you can use [Semgrep CI](../semgrep-ci.md) directly:
```
Comment (pabloest) [inline docs/troubleshooting/gitlab-sast.md:23]: Might want to also add a link to the sample GitLab CI config: (/sample-ci-configs/#gitlab-ci)

File: docs/troubleshooting/rules.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
+# Troubleshooting rules
+
+This page intends to help rule authors fix common mistakes when writing Semgrep rules.
+If you have a problem while running a rule you didn't write yourself, please
+[open a GitHub issue in the semgrep-rules repository](https://github.com/returntocorp/semgrep-rules/issues/new/choose)
+
+[TOC]
+
+# My pattern could not be parsed
```
Comment (pabloest) [inline docs/troubleshooting/rules.md:9]:
```suggestion
# My pattern couldn’t be parsed
```

File: docs/troubleshooting/app.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+[TOC]
+
+# A project is reporting the last scan 'never started'
+
+This status means that your CI job never authenticated to Semgrep App.
+
+Check your CI provider (such as GitHub Actions) for the latest Semgrep job execution.
+
+## If you cannot find a Semgrep CI job
```
Comment (pabloest) [inline docs/troubleshooting/app.md:11]:
```suggestion
## If you can’t find a Semgrep CI job
```

File: mkdocs.yml [hunk 10]
```diff
@@ -34,6 +34,9 @@ nav:
       - Supported languages: status.md
       - Trophy case: trophy-case.md
   - Help:
+      - troubleshooting/app.md
```
Comment (pabloest) [inline mkdocs.yml:37]: I suggest renaming `app.md` to `semgrep-app.md` just to keep closer to the naming convention.

**Review by pabloest — APPROVED** [review]
This is great, @underyx ! Most of my comments are small copy edits.

File: docs/troubleshooting/gitlab-sast.md (hunk 6 above)
Comment (underyx) [reply docs/troubleshooting/gitlab-sast.md:21]: There's no workaround that I could find :(

**Pull request merged**